boskos/client/client.go

```diff
@@ -93,6 +94,28 @@ func (c *Client) Acquire(rtype, state, dest string) (*common.Resource, error) {
 	return r, nil
 }
 
+// AcquireWait blocks until Acquire returns the specified resource or the
+// provided context is cancelled or its deadline exceeded.
+func (c *Client) AcquireWait(ctx context.Context, rtype, state, dest string) (*common.Resource, error) {
+	// Try to acquire the resource until available or the context is
+	// cancelled or its deadline exceeded.
+	for {
+		r, err := c.Acquire(rtype, state, dest)
+		if err != nil {
+			if err == ErrAlreadyInUse || err == ErrNotFound {
+				if ctx != nil && ctx.Err() == nil {
+					time.Sleep(time.Second * 3)
```

Comment:
can you something like:
```golang
select {
	case <-ctx.Done():
		return nil, ctx.Err() or err
	case <-time.After(3 * time.Second):
		continue
}
```

Reply:
Hi @sebastienvas,

I like it. I'll likely return `err` as it could be an expected `Err` from the package.

Reply:
Hi @sebastienvas,

Ah, now I remember why I *didn't* use a `select` -- I'm allowing `ctx` to be `nil`. I'll just set it to `context.Background()` if it comes in `nil`. I prefer the `select` approach.